hs_geo_raster_resource/migrations/custom_migration_for_raster_meta_update_20160512.py

```diff
@@ -0,0 +1,139 @@
+                        if band_obj:
+                            res.metadata.update_element('bandInformation',
+                                                        band_obj.id,
+                                                        maximumValue=band_meta['maximumValue'],
+                                                        minimumValue=band_meta['minimumValue'],
+                                                        noDataValue=band_meta['noDataValue'],
+                                                        )
+                            is_modified = True
+
+                # resource modify update
+                if is_modified:
+                    bag_name = 'bags/{res_id}.zip'.format(res_id=res.short_id)
+                    if istorage.exists(bag_name):
+                        # delete the resource bag as the old bag is not valid
+                        istorage.delete(bag_name)
+                    resource_modified(res, res.creator)
+
+                    success += 1
+                    log.info('Raster metadata is updated successfully for resource:ID:{} '
+                             'Title:{}'.format(res.short_id, res.metadata.title.value))
+                else:
+                    log.info('Raster metadata is not updated for resource:ID:{} '
+                             'Title:{}'.format(res.short_id, res.metadata.title.value))
+                    not_modify.append({res.short_id: res.metadata.title.value})
+
+            except Exception as e:
+                fail.append({res.short_id : res.metadata.title.value})
+                log.info('Raster metadata update failed for resource:ID:{} '
+                        'Title:{}'.format(res.short_id, res.metadata.title.value))
+
+    print 'total resources: {}'.format(str(RasterResource.objects.all().count()))
+    print 'success update resources: {}'.format(str(success))
+    print 'failed update resources: {}'.format(str(fail))
+    print 'not updated resources: {}'.format(str(not_modify))
+
+
+def undo_migrate_tif_file(apps, schema_editor):
+
+    # loop through all the band information and change the nodata value, min, max values as none
+    for res in RasterResource.objects.all():
+        try:
+            for band_obj in res.metadata.bandInformation:
+                res.metadata.update_element('bandInformation',
+                                            band_obj.id,
+                                            maximumValue=None,
+                                            minimumValue=None,
+                                            noDataValue=None,
+                                            )
+        except:
+            pass
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('hs_geo_raster_resource', '0005_auto_20160509_2116'),
+    ]
+
+    operations = [
+        migrations.RunPython(code=migrate_tif_file, reverse_code=undo_migrate_tif_file),
+    ]
```

Comment:
@hyi I think Django can check if a file has been executed for migration so that the file won't be run again.